File: Prácticas/java/src/deepspace/PowerEfficientSpaceStation.java
/**
 *
 * @author elena
 */

package deepspace;

public class PowerEfficientSpaceStation extends SpaceStation {
    private static final float EFFICIENCYFACTOR= 1.1f;
    
    PowerEfficientSpaceStation(SpaceStation station){
        super(station);
    }
    
    @Override
    public float fire(){
        return super.fire()*EFFICIENCYFACTOR;
    }
    
    @Override
    public float protection(){
        return super.protection()*EFFICIENCYFACTOR;
    }
    
    @Override
    public Transformation setLoot(Loot l){
        if(super.setLoot(l) == Transformation.GETEFFICIENT)
            return Transformation.GETEFFICIENT;
        else return Transformation.NOTRANSFORM;
    }
    
    public PowerEfficientSpaceStationToUI getUIversion(){
        return new PowerEfficientSpaceStationToUI(this);
    }
}

File: Prácticas/ruby/lib/PowerEfficientSpaceStation.rb
require_relative 'SpaceStation'
require_relative 'Transformation'
require_relative 'PowerEfficientSpaceStationToUI'

module Deepspace
  class PowerEfficientSpaceStation < SpaceStation
    @@EFFICIENCYFACTOR= 1.1.freeze
    
    def initialize(station)
      # super.newCopy(st)  o bien:  super(st.name,st.getSupplies, st.nMedals, st.weapons,st.shieldBoosters,st.hangar,st.pendingDamage)
      s = SuppliesPackage.new(station.ammoPower, station.fuelUnits, station.shieldPower)
     super(station.name, s, station.weapons, station.shieldBoosters, station.nMedals)
    end
    
    def fire
      super * @@EFFICIENCYFACTOR
    end
    
    def protection
      super * @@EFFICIENCYFACTOR
    end
    
    def setLoot(l)
      if super==Transformation::GETEFFICIENT
        return Transformation::GETEFFICIENT
      else
        return Transformation::NOTRANSFORM
      end
    end
    
    def getUIversion
      PowerEfficientSpaceStationToUI.new(self)
    end
  end
end
